crates/ruff_python_formatter/README.md

```diff
@@ -0,0 +1,306 @@
+
+Options:
+      --check
+          Avoid writing any formatted files back; instead, exit with a non-zero status code if any files would have been modified, and zero otherwise
+  -o, --output-file <OUTPUT_FILE>
+          Specify file to write the formatter output to (default: stdout)
+      --target-version <TARGET_VERSION>
+          The minimum Python version that should be supported [possible values: py37, py38, py39, py310, py311, py312]
+      --config <CONFIG>
+          Path to the `pyproject.toml` or `ruff.toml` file to use for configuration
+  -h, --help
+          Print help
+
+File selection:
+      --respect-gitignore  Respect file exclusions via `.gitignore` and other standard ignore files
+      --force-exclude      Enforce exclusions, even for paths passed to Ruff directly on the command-line
+
+Miscellaneous:
+      --isolated                         Ignore all configuration files
+      --stdin-filename <STDIN_FILENAME>  The name of the file when passing it through stdin
+
+Log levels:
+  -v, --verbose  Enable verbose logging
+  -q, --quiet    Print lint violations, but nothing else
+  -s, --silent   Disable all logging (but still exit with status code "1" upon detecting lint violations)
+```
+
+Note: `ruff format` is currently hidden by default and will not be visible when running
+`ruff --help`.
+
+Similar to Black, running `ruff format /path/to/file.py` will format the given file or directory
+in-place, while `ruff format --check /path/to/file.py` will avoid writing any formatted files back,
+instead exiting with a non-zero status code if any files are not already formatted.
+
+In future releases, the Ruff formatter will be integrated into `ruff check`.
+
+### VS Code
+
+As of `v2023.34.0`,
+the [Ruff VS Code extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
+ships with support for the Ruff formatter. To enable formatting capabilities, set the
+`ruff.enableExperimentalFormatter` setting to `true` in your `settings.json`, and mark the Ruff
+extension as your default Python formatter:
+
+```json
+{
+  "ruff.enableExperimentalFormatter": true,
+  "[python]": {
+    "editor.defaultFormatter": "charliermarsh.ruff"
+  }
+}
+```
+
+From there, you can format a file by running the `Format Document` command, or enable formatting
+on-save by adding `"editor.formatOnSave": true` to your `settings.json`:
+
+```json
+{
+  "ruff.enableExperimentalFormatter": true,
+  "[python]": {
+    "editor.defaultFormatter": "charliermarsh.ruff",
+    "editor.formatOnSave": true
+  }
+}
+```
+
+### Configuration
+
+The Ruff formatter respects Ruff's [`line-length`](https://beta.ruff.rs/docs/settings/#line-length)
+setting, which can be provided via a `pyproject.toml` or `ruff.toml` file, or on the CLI, as with
+`ruff check`.
+
+In future releases, the Ruff formatter will likely support configuration of quote style (single vs.
+double) and indentation (width, and spaces vs. tabs).
```

Comment:
Depending on how complete the list should be:

* missing semicolons
* line ending
* tab width